dotcom-rendering/src/model/enhance-H2s.ts

```diff
@@ -1,15 +1,54 @@
+import { JSDOM } from 'jsdom';
+
+const extractText = (element: SubheadingBlockElement): string => {
+	const frag = JSDOM.fragment(element.html);
+	if (!frag.firstElementChild) return '';
+	return frag.textContent?.trim() ?? '';
+};
+
+/**
+ * We need the ids to be unique so we can navigate to them.
+ * This function checks if the slug already exists and if it does it adds the count to the end of the slug.
+ */
+export const getUnique = (slug: string, array: string[]): string => {
+	if (array.includes(slug)) {
+		const occurenceCount = array.filter(
+			(currentItem) => currentItem == slug,
+		).length;
+		return `${slug}-${occurenceCount}`;
+	}
+	return slug;
+};
+
+const slugify = (text: string) => {
```

Comment:
I always like to leave a link to the source location that code came in from - it's super useful to see what the thinking behind a chunk of code was

Reply:
Good point! i'll add this in